module/localisation/RobotParticleLocalisation/src/RobotModel.cpp

```diff
@@ -62,7 +62,7 @@ namespace localisation {
                                                const Goal::MeasurementType& type,
                                                const FieldDescription& fd) {
 
-        Transform3D Hfw = fieldStateToTransform3D(state);
+        Transform3D Hfw = convert(fieldStateToTransform3D(convert(state)));
```

Comment:
What?

Reply:
You want to do a double conversion here? What?

Reply:
I honestly think this is cleaner than having to add back the arma boilerplate and arma version of that function to `transform.hpp`, just for this one call.

On a related note, like half of the function calls of the functions in `transform.hpp` are not in localisation. I feel like it's probably worth moving `transform.hpp` to `utility/math/matrix` (and renaming the functions which refer to `Transform3D`s by name, while I'm at it). Thoughts?

Reply:
The changes in these localisation files need to be reverted since we are only trying to change behaviour (and related utilities) in this PR. If this involves adding back in arma versions of functions in `transform.hpp` then thats what needs to happen.

We can consider moving `transform.hpp` to another location later on, once all of arma is gone. However, the functions in `transform.hpp` seem to be largely localisation related to field state transforms and tranforms between the 3D world coordinates and the 2D field coordinates

Reply:
Roger that. I added back the functions which were used in the localisation files to `transform.hpp` in one of the commits I just pushed

Reply:
When you're done, you shouldn't need transform.hpp at all, that functionality is literally built into Eigen with it's Affine3d etc code

Reply:
Roger that. I'll add removing it to the plans for after these PRs.